### PR TITLE
[Ldap] Add union types

### PR DESCRIPTION
**Pull request by antoinemaschi**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Extracted from #41424

### CODE REVIEW

File: src/Symfony/Component/Ldap/Adapter/AbstractConnection.php [hunk 1]
```diff
@@ -19,7 +19,7 @@
  */
 abstract class AbstractConnection implements ConnectionInterface
 {
-    protected $config;
+    protected array $config;
```
Comment (jderusse) [inline src/Symfony/Component/Ldap/Adapter/AbstractConnection.php:22]: revert

File: src/Symfony/Component/Ldap/Adapter/AdapterInterface.php [hunk 2]
```diff
@@ -18,29 +18,21 @@ interface AdapterInterface
 {
     /**
      * Returns the current connection.
-     *
-     * @return ConnectionInterface
      */
-    public function getConnection();
+    public function getConnection(): ConnectionInterface;
```
Comment (jderusse) [inline src/Symfony/Component/Ldap/Adapter/AdapterInterface.php:22]: revert

**Review by jderusse — CHANGES_REQUESTED** [review]
Hello, thank you for your help. 

From https://github.com/symfony/symfony/pull/41424#issuecomment-855662517

> Please DO NOT add return types nor types to properties

**Comment by fabpot** [comment]
@antoinemaschi Do you have time to finish this PR?

**Comment by nicolas-grekas** [comment]
Closing as this now contains only changes that we're not ready to make yet. Thanks for opening.

**Pull request closed without merging**